### DISCUSSION TITLE
The cursor randomly changes to box

### DISCUSSION THREAD

**Discussion by signifex** [syl20bnr/spacemacs#15982] [Q&A]
In insert mode, the cursor changes (after a couple of seconds, or by pressing TAB) to a box from the default bar. 
It started to happen after installing ess and lsp layers, but disabling them makes no effect.
Only when I left one layer (treemacs) installed, I cannot reproduce it.

How to fix this, very often confused in the current mode.

Spacemacs cursors also looks normal:

Value:
(("normal" "DarkGoldenrod2" box)
 ("insert" "chartreuse3"
  (bar . 2))
 ("emacs" "SkyBlue2" box)
 ("hybrid" "SkyBlue2"
  (bar . 2))
 ("replace" "chocolate"
  (hbar . 2))
 ("evilified" "LightGoldenrod3" box)
 ("visual" "gray"
  (hbar . 2))
 ("motion" "plum3" box)
 ("lisp" "HotPink1" box)
 ("iedit" "firebrick1" box)
 ("iedit-insert" "firebrick1"
  (bar . 2)))

#### System Info :computer:
- OS: gnu/linux
- Emacs: 27.1
- Spacemacs: 0.999.0
- Spacemacs branch: develop (rev. fd749704f)
- Graphic display: nil
- Running in daemon: nil
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
‘‘‘elisp
(auto-completion
 (auto-completion :variables auto-completion-return-key-behavior ’complete auto-completion-tab-key-behavior ’cycle auto-completion-complete-with-key-sequence nil auto-completion-complete-with-key-sequence-delay 0.1 auto-completion-minimum-prefix-length 2 auto-completion-idle-delay 0.2 auto-completion-private-snippets-directory nil auto-completion-enable-snippets-in-popup nil auto-completion-use-company-box nil auto-completion-enable-snippets-in-popup t auto-completion-enable-help-tooltip t auto-completion-enable-sort-by-usage t)
 python emacs-lisp helm imenu-list ess
 (ess :variables ess-r-backend ’lsp ess-assign-key "\255")
 lsp spell-checking treemacs)

**Comment by lebensterben** [comment]
use gui emacs

**Reply by signifex** [reply]
Why? I feel more comfortable to use Spacemacs without GUI, and it worked fine until new layers were installed.

**Comment by signifex** [comment]
I removed auto-completion and lsp layers, and now spacemacs works fine. Still trying to install them, without this issue.
